pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -179,6 +179,10 @@ func buildSidecarVirtualHostsForVirtualService(
 		// This is a gross HACK. Fix me. Its a much bigger surgery though, due to the way
 		// the current code is written.
 		serviceByPort[80] = nil
```

Comment:
If we deal with all ports then do we still need this port 80 part?

Reply:
same comment

Reply:
even if they don't have a port 80 specified? 

I thought before we have a hack to set 80

Now we correctly set this, and only want 80 if they set match to 80?

I don't really understand this though so maybe that doesn't make sense